internal/topology/variables/clusterclass_variable_validation_test.go

```diff
@@ -631,6 +632,71 @@ func Test_ValidateClusterClassVariable(t *testing.T) {
 				},
 			},
 		},
+		{
+			name:    "fail if field is required below properties and sets a default",
```

Comment:
The test fails at this check: https://github.com/go-openapi/validate/blob/c9f1d67f84a54853d03661cfdff7b2d3238c79c5/object_validator.go#L228

This is the current behaviour of the way we're calling the validation methods.

Reply:
This seems expected to me.

The default value is simply not valid according to the schema, but it has to be

Reply:
This test probably also has another issue which is that "value" is not a defined field

Reply:
> This test probably also has another issue which is that "value" is not a defined field
You're right about this - updated now.

Reply:
>The default value is simply not valid according to the schema, but it has to be

This is also my assumption - let's see if we hear from the person who reported this and otherwise we can just merge this as an additional test - or not if you think it's overkill.

Reply:
I'm fine with adding more coverage around tricky cases like this